transport/src/test/java/io/netty/bootstrap/ServerBootstrapTest.java

```diff
@@ -59,4 +66,74 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
             group.shutdownGracefully();
         }
     }
+
+    @Test(timeout = 3000)
+    public void testParentHandler() throws Exception {
+        testParentHandler(false);
+    }
+
+    @Test(timeout = 3000)
+    public void testParentHandlerViaChannelInitializer() throws Exception {
+        testParentHandler(true);
+    }
+
+    private static void testParentHandler(boolean channelInitializer) throws Exception {
+        final LocalAddress addr = new LocalAddress("test");
```

Comment:
should we just use `InetSocketAddress(0)` so there is less chance of collision with other tests?